From 8eaa5f115df83ca9486141ab6334530f45d59fbe Mon Sep 17 00:00:00 2001
From: Chris Brandt <chris.brandt@renesas.com>
Date: Thu, 9 Jun 2022 18:06:43 -0400
Subject: [PATCH 2/4] drm: rzg2l_mipi_dsi: initialize panel after high speed
 clock setup

If the panel requires DCS commands, this needs to be done after the high
speed clock is configured. Otherwise you might get a timeout error when
sending DCS commands.
---
 drivers/gpu/drm/rcar-du/rzg2l_mipi_dsi.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rzg2l_mipi_dsi.c b/drivers/gpu/drm/rcar-du/rzg2l_mipi_dsi.c
index e4aeaceb12f7..c745bc99ccbf 100644
--- a/drivers/gpu/drm/rcar-du/rzg2l_mipi_dsi.c
+++ b/drivers/gpu/drm/rcar-du/rzg2l_mipi_dsi.c
@@ -616,17 +616,17 @@ static void rzg2l_mipi_dsi_enable(struct drm_bridge *bridge)
 	if (ret < 0)
 		return;
 
-	if (mipi_dsi->panel) {
-		drm_panel_prepare(mipi_dsi->panel);
-		drm_panel_enable(mipi_dsi->panel);
-	}
-
 	rzg2l_mipi_dsi_set_display_timing(mipi_dsi);
 
 	ret = rzg2l_mipi_dsi_start_hs_clock(mipi_dsi);
 	if (ret < 0)
 		return;
 
+	if (mipi_dsi->panel) {
+		drm_panel_prepare(mipi_dsi->panel);
+		drm_panel_enable(mipi_dsi->panel);
+	}
+
 	ret = rzg2l_mipi_dsi_start_video(mipi_dsi);
 	if (ret < 0)
 		return;
-- 
2.28.0

